pkg/ui/react-app/src/thanos/pages/blocks/helpers.ts

```diff
@@ -103,3 +103,28 @@ export const download = (blob: Block): string => {
 
   return url;
 };
+
+export const getOverlappingBlocks = (blockPools: { [source: string]: BlocksPool }): Set<string> => {
+  let result: Set<string> = new Set();
+
+  Object.keys(blockPools).forEach((pk) => {
+    let blocks = Object.values(blockPools[pk]);
+    for (let i = 0; i < blocks.length; i++) {
+      if (blocks[i].length <= 1) {
+        continue;
+      }
+      for (let j = 0; j < blocks[i].length - 1; j++) {
+        for (let k = 0; k < blocks[i][j].length; k++) {
+          for (let l = 0; l < blocks[i][j + 1].length; l++) {
```

Comment:
Can we really assume that we only need to compare with the next row? Or do we need to compare against all blocks in all other rows?